src/JsonSchema/TypeFactory.php

```diff
@@ -50,14 +63,40 @@ public function setSchemaFactory(SchemaFactoryInterface $schemaFactory): void
     public function getType(Type $type, string $format = 'json', ?bool $readableLink = null, ?array $serializerContext = null, Schema $schema = null): array
     {
         if ($type->isCollection()) {
-            $subType = new Type($type->getBuiltinType(), $type->isNullable(), $type->getClassName(), false);
-
-            return [
-                'type' => 'array',
-                'items' => $this->getType($subType, $format, $readableLink, $serializerContext, $schema),
-            ];
+            $keyType = $type->getCollectionKeyType();
+            $subType = $type->getCollectionValueType()
```

Comment:
CS: keep in one line

Reply:
Disagree: stated above in https://github.com/api-platform/core/pull/3402#discussion_r381574437

Reply:
@alanpoulain asked for the same thing in the previous comment. I think we'd like to keep our CS.

Reply:
If the custom CS is really the blocker for this patch: edaac898c8c741a4b425acaaa85da86bf8dd94ec